### DISCUSSION TITLE
Angular Swiper - localized styling on swiper-slide

### DISCUSSION THREAD

**Discussion by pangyikhei** [nolimits4web/swiper#5277] [Q&A]
It seems that if I add an Angular swiper component to an Angular component that has the default encapsulation level, none of the styling in my component's stylesheet will apply to the swiper-slide elements.

ex.
````
.swiper-slide {
  height: 50px;
}
````

It only gets applied if I change the encapsulation to `ViewEncapsulation.None`, but I would like for the changes to be localized to the component.

Note: adding a new class to the `<ng-template swiperSlide [class]="'newCls'">` also doesn't work if the class is defined in the component's stylesheet.

Is there a better way to do this? (short of grabbing the element ref and setting the style directly)

**Comment by MatejaSpindler** [comment]
use ::ng-deep with .swiper-slide class

In your style component you should write:

:host{
::ng-deep .swiper-slide{
height: 50px
}
}

::ng-deep work for me...

**Reply by pangyikhei** [reply]
Is there a another method without using the deprecated ::ng-deep?
